### PR TITLE
fix(app): home all axes after lw calibration to allow deck access

### LINKED ISSUE #4034: PCR Seal Prompt Gantry Move
## overview

At the end of labware calibration, when a TC is present, the user is prompted to add the automation seal to the heated lid. Currently, this prompt  occurs while the gantry is still over A1 of the TC's contained plate.  This gets in the way of putting the seal on.

We need to allow them to move the gantry clear or return tip from this modal.

This has implications on the "Return Tip and Proceed to Run" button.  If the PCR modal appears, the later button should probably not say "Return Tip"

### PR DESCRIPTION
**Pull request by b-cooper**
## Overview

After returning the tip used in lw calibration, the gantry remains above the tiprack. This makes
accessing the items on the deck difficult if need be before the run. It is especially confusing in
the case of a TC protocol, as the user is prompted to place the PCR seal on the module, while the
gantry may be physically blocking it. This appends the behavior of the final "return tip" command of
lw calibration to include a "home all axes" command as well.

Closes #4034

## Review Requests

 - [ ] gantry should home when "Return Tip and Proceed to Run" is pressed at the end of lw calibration for basic protocol

 - [ ] gantry should home when "Return Tip and Proceed to Run" is pressed at the end of lw calibration for protocol that involves the Thermocycler, and you should be prompted to place PCR seal on the TC lid after the home is complete.

### CODE REVIEW

**Comment by SyntaxColoring** [comment]
The OT-2 also homes at the start of the actual protocol run. Can we skip that now, now that homing is done here? (As long as labware calibration isn't skipped.)

File: api/src/opentrons/hardware_control/__init__.py [hunk 1]
```diff
@@ -452,6 +452,8 @@ def halt(self):
         """
         self._log.info("Halting")
         self._backend.hard_halt()
+        # call on attached modules "halt" func that cancels any waits etc.
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/__init__.py]: intentionally checked in?

Comment (b-cooper) [reply api/src/opentrons/hardware_control/__init__.py]: Nope, I thought I removed that, sorry

**Comment by b-cooper** [comment]
@SyntaxColoring, you're right in saying that homing already occurs at the beginning of the protocol.  We can't assume that users have performed labware calibration so we can't remove that home call. In the case of a user running the protocol right after calibration, the second home will be much quicker as axes will already be near the end stop. Though this is not the ideal sequence of events for the UI, until the RPC session management is rewritten and the calibration UI is updated, this is the least invasive and most secure solution.

File: api/src/opentrons/api/calibration.py [hunk 2]
```diff
@@ -259,6 +259,13 @@ def home(self, instrument):
             inst.home()
         self._set_state('ready')
 
+    @_require_lock
+    def home_all(self):
+        log.info('Homing via Calibration Manager')
+        self._set_state('moving')
+        self._hardware.home()
```
Comment (sfoster1) [inline api/src/opentrons/api/calibration.py]: we should clear the location cache here, and also possibly just use the protocol context home

File: app/src/components/CalibrateLabware/ProceedToRun.js [hunk 3]
```diff
@@ -46,38 +49,50 @@ function InfoBoxButton(props: Props) {
...
-                className={styles.place_seal_image}
-                alt="place rubber PCR seal on lid of Thermocycler Module"
-              />
-            </div>
-            <PrimaryButton
-              className={styles.open_lid_button}
-              onClick={() => {
-                dispatch(push(`/run`))
-              }}
+          {inProgress && (
```
Comment (mcous) [inline app/src/components/CalibrateLabware/ProceedToRun.js]: Should this condition be a ternary rather than two separate boolean checks?

```js
{runPrepModalOpen ? (
  <ModalPage ...
) : (
  <AlertModal ...
)}
```

File: app/src/components/CalibrateLabware/ProceedToRun.js [hunk 4]
```diff
@@ -46,38 +49,50 @@ function InfoBoxButton(props: Props) {
...
+            <ModalPage
+              contentsClassName={cx({
+                [styles.modal_contents]: inProgress,
```
Comment (mcous) [inline app/src/components/CalibrateLabware/ProceedToRun.js]: `styles.modal_contents` is a single rule that says `background-color: white;`. This conflicts directly with `styles.in_progress_contents`, which says `background-color: transparent;`. Which one is correct?

This may be a moot point depending on the `<SpinnerModal>` comment below

File: app/src/components/CalibrateLabware/ProceedToRun.js [hunk 5]
```diff
@@ -46,38 +49,50 @@ function InfoBoxButton(props: Props) {
...
+                [styles.in_progress_contents]: inProgress,
+              })}
             >
-              Confirm PCR Seal is in place
-            </PrimaryButton>
-          </AlertModal>
+              <InProgressContents />
```
Comment (mcous) [inline app/src/components/CalibrateLabware/ProceedToRun.js]: `<InProgressContents />` is an old component, and was designed to show a spinner inside an existing wizard box. As used here, I think you'd be better served by `<SpinnerModal>` in `app/src/components/modals`

File: app/src/robot/reducer/calibration.js [hunk 6]
```diff
@@ -150,6 +153,12 @@ export default function calibrationReducer(
     case 'robot:SET_MODULES_REVIEWED':
       return handleSetModulesReviewed(state, action)
 
+    case 'robot:RETURN_TIP':
```
Comment (mcous) [inline app/src/robot/reducer/calibration.js]: Could you add unit tests to `app/src/robot/test/calibration-reducer.test.js` for this behavior?

File: components/src/modals/ModalPage.js [hunk 7]
```diff
@@ -9,7 +9,7 @@ import styles from './modals.css'
 
 type Props = {|
   /** Props for title bar at top of modal page */
-  titleBar: TitleBarProps,
+  titleBar?: TitleBarProps,
```
Comment (mcous) [inline components/src/modals/ModalPage.js]: `<ModalPage>` exists to be a modal with a title bar. Can you revert this change and either:
a) Use `<Modal>` instead or
b) As suggested above, use `<SpinnerModal>`

File: app/src/components/CalibrateLabware/ProceedToRun.js [hunk 8]
```diff
@@ -12,14 +12,17 @@ import pcrSealSrc from '../../img/place_pcr_seal.png'
 import { Portal } from '../portal'
 import styles from './styles.css'
 
+const is_homing_message = 'Returning tip and homing robot'
```
Comment (mcous) [inline app/src/components/CalibrateLabware/ProceedToRun.js]: ALL_CAPS?
